### PR TITLE
Deprecating all C headers

### PR DESCRIPTION
**Pull request by CursedRock17**
This pull request is another and possibly the last needed addition to complete the breakdown of PR #120 in which all C headers are deprecated to C++ headers to solve this issue:
> But there is another problem here, and that is that message_filters isn't a "proper" C++ package. In particular, the header files still have the .h extension, and so our linters consider them to be C headers, not C++ headers

The PR appears to be super messy because I guess renaming files counts as making a whole new file in Git's eyes, but really each old file was renamed to have the `.hpp` tag with the correct header guard, then a new file `.h` was created with a warning message to use the C++ style header, so not too bad.

### CODE REVIEW

File: include/message_filters/cache.h [hunk 1]
```diff
@@ -1,342 +1,22 @@
-// Copyright 2008, Willow Garage, Inc. All rights reserved.
+// Copyright 2024 Open Source Robotics Foundation, Inc.
```
Comment (ahcorde) [inline include/message_filters/cache.h:1]:
```suggestion
// Copyright 2024 Open Source Robotics Foundation, Inc.
// All rights reserved.
```

Comment (ahcorde) [reply include/message_filters/cache.h:1]: to be consistent with the rest of headers, here and in the rest of files

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Comment by ahcorde** [comment]
Updated downstream packages:
 - https://github.com/ros-perception/image_common/pull/320
 - https://github.com/ros-perception/point_cloud_transport/pull/94
 - https://github.com/ros2/geometry2/pull/702
 - https://github.com/ros2/rviz/pull/1239
 - https://github.com/ros-perception/image_pipeline/pull/1012

**Comment by ahcorde** [comment]
I added rviz, geometry2, image_common and point_cloud_transport changes too

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21494/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15777/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/993/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22208/)

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21497/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15779/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/995/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22210/)

**Review by ahcorde — APPROVED** [review]

**Comment by CursedRock17** [comment]
Since this pull request was meant to be the last of pull request #120, I feel we should be able to close both #120 and its respective issue of #29 since linters have been totally added. We just have to revisit this pull request in I believe L-turtle to remove the deprecations.

**Comment by ahcorde** [comment]
> Since this pull request was meant to be the last of pull request #120, I feel we should be able to close both #120 and its respective issue of #29 since linters have been totally added. We just have to revisit this pull request in I believe L-turtle to remove the deprecations.

See this comment https://github.com/ros2/message_filters/pull/120

**Pull request merged**